### ISSUE TITLE
`syncCalendars` marking unchanged calendars as `updated`

### ISSUE THREAD

**Issue by garshythoel** [natelindev/tsdav#200]
I am using the library in the following way to fetch any changes made since the last sync

```
const existingCalendars = await db
    .select()
    .from(calendars)
    .where(
      and(eq(calendars.userId, userId), eq(calendars.accountId, accountId))
    );

  const { created, updated, deleted } = await syncCalendars({
    oldCalendars: existingCalendars.map((cal) => ({
      url: cal.url,
      ctag: cal.ctag,
      syncToken: cal.syncToken,
      displayName: cal.name
    })) as DAVCalendar[],
    detailedResult: true,
    headers: client.authHeaders,
    account: client.account
  });
```

The debug logs show the following
```
  tsdav:calendar new calendars:  +0ms
  tsdav:calendar updated calendars: Formula 1,Birthdays,Todoist,harshyt.goel@gmail.com,House Viewings 2019,Todoist,Family,HZ Holdings +0ms
  tsdav:collection smart collection sync with type caldav and method webdav +0ms
  tsdav:collection smart collection sync with type caldav and method webdav +2ms
  tsdav:collection smart collection sync with type caldav and method webdav +2ms
  tsdav:collection smart collection sync with type caldav and method webdav +0ms
  tsdav:collection smart collection sync with type caldav and method webdav +1ms
  tsdav:collection smart collection sync with type caldav and method webdav +0ms
  tsdav:collection smart collection sync with type caldav and method webdav +0ms
  tsdav:collection smart collection sync with type caldav and method webdav +1ms
  ```
  
  On an immediate refresh - such that there are no updates on the mentioned calendars - i get the same logs again. Marking the unchanged calendars as "updated"
  
here is the code i'm using to check
```
  // Handle updated calendars
  const calendarsToUpdate: Calendar[] = updated.map((cal) => {
    const existingCal = existingCalendars.find((e) => e.url === cal.url);

    console.log(
      'ctag',
      existingCal?.ctag,
      typeof existingCal?.ctag,
      existingCal?.ctag === cal.ctag,
      cal.ctag,
      typeof cal.ctag
    );

    console.log('syncToken', existingCal?.syncToken === cal.syncToken);
    return {
      ...existingCal!,
      name: typeof cal.displayName === 'string' ? cal.displayName : '',
      description: cal.description ?? existingCal!.description,
      timezone: cal.timezone ?? existingCal!.timezone,
      syncToken: cal.syncToken ?? existingCal!.syncToken,
      ctag: cal.ctag ?? existingCal!.ctag,
      lastSynced: new Date()
    };
  });
  ```
  
The console logs show that the c-tag comparison is failing because of a difference in the type received from the syncCalendar calls `number` to what is expected as a part of the input with the typescript definitions `string | null`

```
ctag 63857146129 string false 63857146129 number
syncToken true
ctag 63857487675 string false 63857487675 number
syncToken true
ctag 63847717763 string false 63847717763 number
syncToken true
ctag 63857613174 string false 63857613174 number
syncToken true
ctag 63838654712 string false 63838654712 number
syncToken true
ctag 63833874449 string false 63833874449 number
syncToken true
ctag 63821455767 string false 63821455767 number
syncToken true
ctag 63857485622 string false 63857485622 number
syncToken true
```

This seems like perhaps there should be some inherent coercion to do accurate comparisons. The issue may be here in what is expected vs what we see:
https://github.com/natelindev/tsdav/blob/df2088ac6a5e5017ab35385cb839745141ef55d0/src/calendar.ts#L464

 Let me know if you'd like some more information but some guidance/fix here would be great! I haven't checked if `etag`s have the same issue but is possible - i'm using this with the google endpoints.

Thanks

**Comment by natelindev** [comment]
I'll take a look this week.

**Comment by garshythoel** [comment]
Awesome, thanks! Let me know if I can provide any more info

**Comment by natelindev** [comment]
fixed in v2.1.1
